src/server/game/Movement/MotionMaster.cpp

```diff
@@ -88,9 +88,6 @@ void MotionMaster::UpdateMotion(uint32 diff)
     if (!_owner)
         return;
 
-    if (_owner->HasUnitState(UNIT_STATE_ROOT | UNIT_STATE_STUNNED)) // what about UNIT_STATE_DISTRACTED? Why is this not included?
-        return;
-
```

Comment:
Each movenment decides what to do when those states are detected.

If this is checked to prevent update on MotionMaster, that is no longer an option and its the reason why I removed it.